src/main/java/com/team/buddyya/auth/jwt/JwtUtils.java

```diff
@@ -0,0 +1,87 @@
+package com.team.buddyya.auth.jwt;
+
+import com.team.buddyya.auth.dto.request.TokenInfoRequest;
+import com.team.buddyya.auth.exception.AuthException;
+import com.team.buddyya.auth.exception.AuthExceptionType;
+import io.jsonwebtoken.*;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.security.Key;
+import java.time.ZonedDateTime;
+import java.util.Date;
+
+@Slf4j
+@Component
+public class JwtUtils {
+
+    private final Key key;
+    private final long accessTokenExpireTime;
+    private final long refreshTokenExpireTime;
+
+    public JwtUtils(@Value("${auth.jwt.secret}") String secretKey,
+                    @Value("${auth.jwt.access.expiration}") long accessTokenExpireTime,
+                    @Value("${auth.jwt.refresh.expiration}") long refreshTokenExpireTime
+    ) {
+        byte[] keyBytes = Decoders.BASE64.decode(secretKey);
+        this.key = Keys.hmacShaKeyFor(keyBytes);
+        this.accessTokenExpireTime = accessTokenExpireTime;
+        this.refreshTokenExpireTime = refreshTokenExpireTime;
+    }
+
+    public String createAccessToken(TokenInfoRequest context) {
+        return createToken(context, accessTokenExpireTime);
+    }
+
+    public String createRefreshToken(TokenInfoRequest context) {
+        return createToken(context, refreshTokenExpireTime);
+    }
+
+    private String createToken(TokenInfoRequest context, long expiredTime) {
+        Claims claims = Jwts.claims();
+        claims.put("studentId", context.studentId());
+        ZonedDateTime now = ZonedDateTime.now();
+        ZonedDateTime tokenValidity = now.plusSeconds(expiredTime);
```

Comment:
P3: 불변성을 유지하는데 더 유리한 코드라고 생각합니다!
```suggestion
        ZonedDateTime tokenValidity =  ZonedDateTime.now().plusSeconds(expiredTime);
```

Reply:
now가 뒤에 한번 더 쓰여 따로 변수로 설정하고 재사용하였습니다!